geo/src/main/java/org/odk/collect/geo/geopoint/GeoPointMapActivity.java

```diff
@@ -325,6 +329,22 @@ protected void restoreFromInstanceState(Bundle state) {
         locationStatus.setVisibility(state.getInt(LOCATION_STATUS_VISIBILITY_KEY, View.GONE));
     }
 
+    // https://github.com/getodk/collect/issues/5293
+    @Override
+    public void onBackPressed() {
```

Comment:
`onBackPressed` has been deprecated (see [Android docs](https://developer.android.com/reference/android/app/Activity#onBackPressed()) and [this issue](https://github.com/getodk/collect/issues/5142) for more info), but annoyingly doesn't show up as such when overriding in a class that extends `ComponentActivity` due to a `@SuppressWarnings` 🤦‍♂️.

As the docs say, adding actions that occur when back is pressed should now be done via a `OnBackPressedDispatcher` instance. There's an example of us doing this for an Activity in [CrashHandlerActivity](https://github.com/getodk/collect/blob/b42fd42b3283bdbcea981b6e0a0ce51089669360/collect_app/src/main/java/org/odk/collect/android/activities/CrashHandlerActivity.kt#L25).

Reply:
An interesting example of composition vs overriding - and of Kotlin to Java.